flow/layers/color_filter_layer.cc

```diff
@@ -45,23 +48,29 @@ void ColorFilterLayer::Paint(PaintContext& context) const {
   TRACE_EVENT0("flutter", "ColorFilterLayer::Paint");
   FML_DCHECK(needs_painting(context));
 
-  AutoCachePaint cache_paint(context);
-
   if (context.raster_cache) {
+    AutoCachePaint cache_paint(context);
     if (layer_raster_cache_item_->IsCacheChildren()) {
-      cache_paint.setColorFilter(filter_);
+      cache_paint.setColorFilter(filter_->skia_object());
     }
     if (layer_raster_cache_item_->Draw(context, cache_paint.sk_paint())) {
       return;
     }
   }
 
-  cache_paint.setColorFilter(filter_);
-
-  Layer::AutoSaveLayer save = Layer::AutoSaveLayer::Create(
-      context, paint_bounds(), cache_paint.sk_paint());
-
-  PaintChildren(context);
+  AutoCachePaint cache_paint(context);
+  if (context.leaf_nodes_builder) {
+    cache_paint.setColorFilter(filter_.get());
+    context.leaf_nodes_builder->saveLayer(&paint_bounds(),
+                                          cache_paint.dl_paint());
+    PaintChildren(context);
+    context.leaf_nodes_builder->restore();
+  } else {
+    cache_paint.setColorFilter(filter_ ? filter_->skia_object() : nullptr);
```

Comment:
Same as earlier - use the Dl version of this method so we can delete the sk version.

Reply:
done